tests/neptune/test_artifacts.py

```diff
@@ -0,0 +1,196 @@
+        filename, filepath = fake.file_name(), fake.file_path(depth=3).lstrip('/')
+
+        bucket_name, s3_client = bucket
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with preserve_cwd(tmp):
+
+                with open(filename, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                os.makedirs(Path(filepath).parent, exist_ok=True)
+                with open(filepath, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                s3_client.meta.client.upload_file(filename, bucket_name, filename)
+                s3_client.meta.client.upload_file(filepath, bucket_name, filepath)
+
+        run[first].track_files(f's3://{bucket_name}/')
+
+        run.sync()
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with with_check_if_file_appears(f'{tmp}/{filename}'):
+                run[first].download(tmp)
+
+    def test_s3_existing(self, run: Run, bucket):
+        first, second = self.gen_key(), self.gen_key()
+        filename, filepath = fake.file_name(), fake.file_path(depth=3).lstrip('/')
+
+        bucket_name, s3_client = bucket
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with preserve_cwd(tmp):
+
+                with open(filename, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                os.makedirs(Path(filepath).parent, exist_ok=True)
+                with open(filepath, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                s3_client.meta.client.upload_file(filename, bucket_name, filename)
+                s3_client.meta.client.upload_file(filepath, bucket_name, filepath)
+
+        run[first].track_files(f's3://{bucket_name}/')
+        run[second].track_files(f's3://{bucket_name}/{filename}')
+        run.sync()
+
+        # Track to existing
+        run[second].track_files(f's3://{bucket_name}/{filepath}', destination=str(Path(filepath).parent))
+        run.sync()
+
+        assert run[first].fetch_hash() == run[second].fetch_hash()
+        assert run[first].fetch_files_list() == run[second].fetch_files_list()
+
+    def test_local_existing(self, run: Run):
+        first, second = self.gen_key(), self.gen_key()
+        filename, filepath = fake.file_name(), fake.file_path(depth=3).lstrip('/')
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with preserve_cwd(tmp):
+                print(os.getcwd(), Path(tmp))
+                with open(filename, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                os.makedirs(Path(filepath).parent, exist_ok=True)
+                with open(filepath, 'w', encoding='utf-8') as handler:
+                    handler.write(fake.paragraph(nb_sentences=5))
+
+                run[first].track_files('.')
```

Comment:
could you add some comments before each of these `track_files` (like "add whole dir", "add first file only", "add second file to existing")?